scripts/1.TCGA-MLexample.py

```diff
@@ -72,84 +75,96 @@
         urllib.request.urlretrieve(url, path)
 
 
-# In[7]:
+# In[8]:
 
 get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'expression.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
 
 
-# In[8]:
+# In[9]:
 
 get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
 
 
-# ## Process Mutation Matrix
-
-# In[9]:
+# In[10]:
 
 y = Y[GENE]
 
 
-# In[10]:
+# In[11]:
 
 # The Series now holds TP53 Mutation Status for each Sample
 y.head(6)
 
 
-# In[11]:
+# In[12]:
 
 # Here are the percentage of tumors with NF1
 y.value_counts(True)
 
 
 # ## Set aside 10% of the data for testing
 
-# In[12]:
+# In[13]:
 
 # Typically, this can only be done where the number of mutations is large enough
 X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
 'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.index), len(X_train), len(X_test))
 
 
-# ## Feature transformation: Standardization
+# ## Median absolute deviation feature selection
 
-# In[13]:
+# In[14]:
 
-def scale_df(df):
-    """Standardize colummns by scaling to z-scores"""
-    scaled_matrix = preprocessing.scale(df)
-    return pd.DataFrame(scaled_matrix, index=df.index, columns=df.columns)
+def fs_mad(x, y):
+    mad = lambda a: np.median(np.absolute(a - np.median(a)))
+    score = np.apply_along_axis(mad, 0, x)
+    p = np.array([0.5]*len(score))
+    return score, p
+feature_select = SelectKBest(fs_mad, k=500)  # select the top 500 features with the highest MAD
 
-# Standardize X separately for the holdout dataset
-X_train, X_test = scale_df(X_train), scale_df(X_test)
 
+# ## Define pipeline and Cross validation model fitting
 
-# ## Cross validation model fitting
+# In[15]:
 
-# In[14]:
+# Include loss='log' in param_grid doesn't work with pipeline somehow
+clf = SGDClassifier(random_state=0, class_weight='balanced', loss='log')
```

Comment:
Sounds good to me. 
It seems that the bug come from the decorator @if_delegate_has_method(delegate='estimator') from the BaseSearchCV class
```python
    @if_delegate_has_method(delegate='estimator')
    def predict_proba(self, X):
        """Call predict_proba on the estimator with the best found parameters.

        Only available if ``refit=True`` and the underlying estimator supports
        ``predict_proba``.

        Parameters
        -----------
        X : indexable, length n_samples
            Must fulfill the input assumptions of the
            underlying estimator.

        """
        return self.best_estimator_.predict_proba(X)
```
If I remove the decorator, everything works as expected:
```
pipeline.predict_proba(X)
Out[4]: 
array([[ 0.42683613,  0.57316387],
       [ 0.28014646,  0.71985354],
       [ 0.29039856,  0.70960144],
       ..., 
       [ 0.72043756,  0.27956244],
       [ 0.9025268 ,  0.0974732 ],
       [ 0.62758795,  0.37241205]])
```